lib/components/unit_button.dart

```diff
@@ -168,13 +196,25 @@ class _UnitButtonState extends State<UnitButton> {
               fit: BoxFit.cover,
             );
     }
-    return Center(
-        child: Text(widget.text,
-            style: new TextStyle(
-                color: widget.highlighted || !widget.primary
-                    ? Colors.white
-                    : Theme.of(context).primaryColor,
-                fontSize: fontSize)));
+    return int.tryParse(widget.text) == null || int.tryParse(widget.text) == 0
```

Comment:
Better to do tryParse once and keep result in an int and use that instead of doing tryParse 3 times